### PR TITLE
Issue 197 bek visibilty options

### PR DESCRIPTION
**Pull request by nhoobin**

### CODE REVIEW

**Comment by nhoobin** [comment]
Referring to https://github.com/catalyst/moodle-quizaccess_seb/issues/197, 



- [x] When this option is disabled, then the text box "Allowed Browser Exam Keys" isn't displayed. The option "Enable quitting of SEB" and the field "Quit password" are displayed.
- [x] When this option is disabled, then the plugin uses the entered quit password and the setting "Enable quitting of SEB".
- [x] When this option is enabled, then display the text box "Allowed Browser Exam Keys" and hide the option "Enable quitting of SEB" and the field "Quit password".
- [ ] Optionally a message could be displayed "The quit password from the uploaded config is used"
- [ ] If the option "Use Browser Exam Keys" is enabled, then the uploaded config file is sent unmodified to the SEB browser.
- [x] If the user attempts to save settings without entering a Browser Exam Key, the plugin should notify the user that at least one is required.

We still modify the uploaded config file as we want to set the start URL to the current quiz, is that still the use case here?

File: classes/settings_provider.php [hunk 1]
```diff
@@ -907,11 +917,11 @@ public static function get_seb_settings_map() : array {
                 'seb_allowuserquitseb' => [
                     'seb_quitpassword' => [],
                 ],
-                'seb_allowedbrowserexamkeys' => [],
```
Comment (nhoobin) [inline classes/settings_provider.php:910]: I also removed these, which I think was wrong, yet these are not seb settings.

File: classes/quiz_settings.php [hunk 2]
```diff
@@ -402,7 +407,10 @@ private function process_bool_settings() {
      */
     private function process_quit_password_settings() {
         $settings = $this->to_record();
-        if (!empty($settings->quitpassword) && is_string($settings->quitpassword)) {
+
+        // If Use Browser Exam Keys is enabled, then we will not process the quit password.
+        if (empty($settings->usebrowserexamkeys) &&
+            !empty($settings->quitpassword) && is_string($settings->quitpassword)) {
```
Comment (dmitriim) [inline classes/quiz_settings.php:413]: do you know, why are we checking this? `is_string($settings->quitpassword)`

File: classes/settings_provider.php (hunk 1 above)
Comment (dmitriim) [reply classes/settings_provider.php:910]: I'd leave it here. As this get_seb_settings_map gives us a good overview of how settings are apply to the form. They are not all SEB settings. See seb_suppresssebdownloadlink . Maybe we should rename the method

File: rule.php [hunk 3]
```diff
@@ -150,6 +150,18 @@ public static function validate_settings_form_fields(array $errors,
             }
         }
 
+        // Edge case for Use Browser Exam Keys and empty Allowed Browser Exam Keys.
+        $cases = [
+            settings_provider::USE_SEB_UPLOAD_CONFIG,
+            settings_provider::USE_SEB_CLIENT_CONFIG
+        ];
+        $requiresetting = $quizsettings->get('requiresafeexambrowser');
+        if (in_array($requiresetting, $cases)) {
```
Comment (dmitriim) [inline rule.php:159]: how about to abstract it to a method + cover with tests?

Comment (dmitriim) [reply rule.php:159]: Not sure about the best place to put it. Maybe to settings_provider?

Comment (dmitriim) [reply rule.php:159]: Also I think we need to empty allowedbrowserexamkeys in quiz_settings as we do with templateid. just to make sure we don't save hidden value from the form

**Comment by dmitriim** [comment]
Closed as agreed not to proceed with these changes.

**Pull request closed without merging**